library/src/cache/patch_manager.rs

```diff
@@ -346,6 +372,19 @@ impl PatchManager {
 }
 
 impl ManagePatches for PatchManager {
+    fn on_init(&mut self) -> Result<()> {
+        // If we were booting a patch but never recorded a successful boot, we assume that
+        // the patch failed to boot. Attempt to fall back.
+        // TODO: this should record a PatchInstallFailure event. https://github.com/shorebirdtech/updater/issues/188
+        if let Some(failed_boot_patch) = self.patches_state.currently_booting_patch.clone() {
```

Comment:
Would our previous logic have recorded an event?

Reply:
Nope. This isn't a regression, so I decided to fix it in a followup PR.

Reply:
Oh wow.  So how does failure reporting ever work then?  I guess only failures where the failure type is "can't parse this snapshot" or similar are reported and any "crash during dart init" are just invisible to users?

Reply:
We submit events for failures that are reported via the `report_launch_failure`, which comes in through the Flutter engine. But yeah, we're potentially dropping a lot of failure events here.